hypervisor/arch/x86/notify.c

```diff
@@ -35,7 +35,8 @@ void smp_call_function(uint64_t mask, smp_call_func_t func, void *data)
 
 	/* wait for previous smp call complete, which may run on other cpus */
 	while (atomic_cmpxchg64(&smp_call_mask, 0UL, mask & INVALID_BIT_INDEX));
-	while ((pcpu_id = ffs64(mask)) != INVALID_BIT_INDEX) {
+	pcpu_id = ffs64(mask);
+	while (pcpu_id != INVALID_BIT_INDEX) {
```

Comment:
for this kind of issues, previously we solve it using:
for (pcpu_id = ffs64(mask); pcpu_id != INVALID_BIT_INDEX; pcpu_id = ffs64(mask) {
}

can this solve the problem? it is just one LOC.

Reply:
The static checker requires that the loop incrementation should be simple, like ++ or -- operators.
If we use 'pcpu_id = ffs64(mask)' as the loop incrementation. The static checker will report the violation.